nifi-nar-bundles/nifi-jwt-bundle/nifi-jwt-processors/src/main/java/org/apache/nifi/processors/jwt/SignJWT.java

```diff
@@ -0,0 +1,165 @@
+@CapabilityDescription("JWT Signs a JSON file with a specified private key. The JSON must be key value pairs.")
+@WritesAttributes({
+    @WritesAttribute(attribute = "failure.reason", description = "If the flow file is routed to the failure relationship "
+            + "the attribute will contain the error message resulting from the validation failure.")
+})
+@SeeAlso({ UnsignJWT.class })
+public class SignJWT extends AbstractProcessor {
+
+    public static final PropertyDescriptor PRIVATE_KEY_PATH = new PropertyDescriptor.Builder()
+            .name("PRIVATE_KEY_PATH").displayName("Private Key").description("Path to private key to sign the JWT. The key must be readable by the NiFi user")
+            .required(true).addValidator(CustomValidators.PRIVATE_KEY_VALIDATOR).build();
+
+    public static final Relationship SUCCESS_REL = new Relationship.Builder().name("success")
+            .description("A FlowFile is routed to this relationship after its JWT has been validated and unsigned")
+            .build();
+
+    public static final Relationship FAILURE_REL = new Relationship.Builder().name("failure").description(
+            "A FlowFile is routed to this relationship if the JWT cannot be validated or unsigned for any reason")
+            .build();
+
+    public static final String FAILURE_REASON_ATTR = "failure.reason";
+    public static final String INVALID_JSON = "Invalid JSON: ";
+    public static final String ERROR_PREFIX = "Error signing JWT: ";
+
+    private PrivateKey privateKey;
+
+    private List<PropertyDescriptor> descriptors;
+    private Set<Relationship> relationships;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+        final List<PropertyDescriptor> descriptors = new ArrayList<PropertyDescriptor>();
+        descriptors.add(PRIVATE_KEY_PATH);
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<Relationship>();
+        relationships.add(SUCCESS_REL);
+        relationships.add(FAILURE_REL);
+        this.relationships = Collections.unmodifiableSet(relationships);
+
+        if (Security.getProvider("BC") == null) {
+            Security.addProvider(new BouncyCastleProvider());
+        }
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        final File key = new File(context.getProperty(PRIVATE_KEY_PATH).getValue());
+        KeyProvider keyProvider;
+        try (final SSHClient sshClient = new SSHClient()) {
+            keyProvider = sshClient.loadKeys(key.getAbsolutePath());
+            privateKey = keyProvider.getPrivate();
+        } catch (IOException e) {
+            privateKey = null;
```

Comment:
I would suggest that failing to obtain the private key should prevent the processor from starting rather than setting this value as a sentinel and yielding during processing.

Reply:
It does with the validationContext set on the property